Cython/Shadow.pyi

```diff
@@ -267,6 +267,51 @@ class typedef(CythonType, Generic[_T]):
 
 NULL: pointer[Any]
 
+# Lock types for synchronization
+class pymutex:
+    """
+    A low-cost mutex lock type available on Python 3.13+.
```

Comment:
This isn't entirely true because we have a backport that uses `PyThead_type_lock` internally.
```suggestion
    A low-cost mutex lock type.
    In Python 3.13+, it uses the fast PyMutex implementation. In older Python versions,
    it falls back to the heavier "PyThread_type_lock".
```